### ISSUE TITLE
Conditional Constraint on Boolean and Hardcoding of BigInteger

### ISSUE THREAD

**Issue by johspaeth** [CROSSINGTUD/Crypto-API-Rules#35]
For BouncyCastle I would like to specify `RSAKeyParameters` in the following way

```
SPEC org.bouncycastle.crypto.params.RSAKeyParameters

OBJECTS
	boolean isPrivate;
	java.math.BigInteger exponent;
	java.math.BigInteger modulus;
	
EVENTS
	c : RSAKeyParameters(isPrivate, modulus, exponent); 

ORDER
	c	

CONSTRAINTS
	isPrivate == true => notHardCoded(exponent) 

ENSURES
	generatedRSAKeyParams[this];
```
Generally, I would like to say: If one uses a RSAKeyParameter with mode isPrivate, the exponent may not be hardcoded within the code. That's what I tried in the `CONSTRAINTS` section. There are two issues:

Firstly, one cannot constraint on a boolean value, i.e., isPrivate == true leads to a compiler error "Couldn't resolve supertype true"
Secondly, we would need to add the special keyword `notHardCoded` which warns when the analysis detects `new BigInteger(..)`(String, ...?) 

Speaking in code, I would like to have warnings as follows

```
BigInteger mod = new BigInteger("a0b8e8321b041acd40b7", 16);
BigInteger pub = new BigInteger("9f0783a49...da", 16);	
BigInteger pri = new BigInteger("21231...cda7", 16); 
RSAKeyParameters privParameters = new RSAKeyParameters(true, mod, pri); <--- warning here
RSAKeyParameters pubParameters = new RSAKeyParameters(false, mod, pub); <--- but no warning here
```
